core-backup/src/test/scala/io/aiven/guardian/kafka/backup/MockedBackupClientInterface.scala

```diff
@@ -84,10 +87,23 @@ class MockedBackupClientInterface(override val kafkaClientInterface: MockedKafka
     */
   override type BackupResult = Done
 
-  override type State = Nothing
+  override type State = Unit
```

Comment:
This had to be changed from `Nothing` to `Unit` because as per design of Scala's type system, no value can have type `Nothing` and with these changes its the first time we have to return a `State` value (so we just use `Unit`).